### PR TITLE
Scannergates 2.0

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Scanner Gates can now have assemblies attached to them! Allow you to trigger anything you want with them!

XGWvtJbPH9

There are 5 wires on them now.
Successful Scan
Failed Scan
Toggle Signaling
Disable Signaling
Enabling Signaling

## Why It's Good For The Game
Imagine the possibilities! Traps! Security Checkpoints! 
 

## Changelog
:cl:
add: Scanner Gates can now trigger assemblies!
/:cl:

### CODE REVIEW

File: code/game/machinery/scan_gate.dm [hunk 1]
```diff
@@ -164,6 +180,15 @@
 		beep = !beep
 	if(beep)
 		alarm_beep()
+		if(!ignore_signals)
+			color = wires.get_color_of_wire(WIRE_FAIL)
+			var/obj/item/assembly/S = wires.get_attached(color)
+			S.activate()
```
Comment (ike709) [inline code/game/machinery/scan_gate.dm]:
```suggestion
			if(istype(S))
			    S.activate()
```

File: code/game/machinery/scan_gate.dm [hunk 2]
```diff
@@ -164,6 +180,15 @@
...
 		alarm_beep()
+		if(!ignore_signals)
+			color = wires.get_color_of_wire(WIRE_FAIL)
+			var/obj/item/assembly/S = wires.get_attached(color)
+			S.activate()
+	else
+		if(!ignore_signals)
+			color = wires.get_color_of_wire(WIRE_PASS)
+			var/obj/item/assembly/S = wires.get_attached(color)
+			S.activate()
```
Comment (ike709) [inline code/game/machinery/scan_gate.dm]:
```suggestion
          if(istype(S))
			    S.activate()

```

**Comment by ghost** [comment]
@ike709 I did as you ask. 👀

**Review by francinum — DISMISSED** [review]
My reviews don't even count

**Comment by ghost** [comment]
@francinum They count to me

**Comment by PowerfulBacon** [comment]
Wow, nice.
Make sure you update the hacking page on our wiki.

**Comment by ike709** [comment]
@francinum @PowerfulBacon neither of you pointed out that this doesn't even compile

**Comment by PowerfulBacon** [comment]
Oh they used spaces instead of tabs?
I was on mobile when I looked at this and didnt scroll down.
Anyway, it doesn't compile  @axanthic

**Comment by ghost** [comment]
> @francinum @PowerfulBacon neither of you pointed out that this doesn't even compile

My bad guys - Thing I messed up on the Push will fix in a bit

**Comment by ghost** [comment]
> @francinum @PowerfulBacon neither of you pointed out that this doesn't even compile

Ike broke it with his cool changes >:C I take 0 responsibility for this. Anyhow should be fixed and set to go.

**Comment by ghost** [comment]
> Wow, nice.
> Make sure you update the hacking page on our wiki.

Is this something I have to do?
I'm unsure what changes should even be made since as far as I'm aware the wire setup is unique to this device. So I don't know if having a specific section just for it makes sense?

**Comment by ghost** [comment]
I'm a dumbass - Should be set to go.

**Comment by ghost** [comment]
Updated the names of the variables and added a general trigger wire.

**Comment by ike709** [comment]
What if the wires are cut?

File: code/game/machinery/scan_gate.dm [hunk 3]
```diff
@@ -160,11 +177,26 @@
 				if(H.nutrition >= detect_nutrition && detect_nutrition == NUTRITION_LEVEL_FAT)
 					beep = TRUE
 
+	if(!ignore_signals)
+		color = wires.get_color_of_wire(WIRE_PROCEED)
+		var/obj/item/assembly/S = wires.get_attached(color)
```
Comment (qwertyquerty) [inline code/game/machinery/scan_gate.dm]: You do these 3 lines 3 times in 3 different places. Just do them once at the end.

File: code/game/machinery/scan_gate.dm [hunk 4]
```diff
@@ -50,12 +55,18 @@
 	else
 		. += "<span class='notice'>The control panel is unlocked. Swipe an ID to lock it.</span>"
 
+/obj/machinery/scanner_gate/update_icon_state()
+	if(panel_open)
```
Comment (qwertyquerty) [inline code/game/machinery/scan_gate.dm]: What is the point of this

**Review by qwertyquerty — CHANGES_REQUESTED** [review]

File: code/game/machinery/scan_gate.dm (hunk 3 above)
Comment (ghost) [reply code/game/machinery/scan_gate.dm]: I'll condense this - it made more sense when there was only 2.

File: code/game/machinery/scan_gate.dm (hunk 4 above)
Comment (ghost) [reply code/game/machinery/scan_gate.dm]: I can remove this or we can leave it if someone plans to implement a maintenance mode sprite

**Comment by ghost** [comment]
> What if the wires are cut?

For other machines if wires are cut does that prevent the ability for items to be pulsed by attachments?
On the scanner it doesn't allow the device to be opened if it's locked so that wouldn't be a good way of sabotaging it.

**Comment by ghost** [comment]
Changes made - Needs to be tested.
Condensed some of the code a bit better. Removed the icon state check - Will re-add if anyone wants to make a sprite for it.

**Comment by Axanthic** [comment]
Closing - Was PermaBanned

**Comment by imw-Oculus** [comment]
Wonderful. A follow-up when this is merged would be to create more variables that the Scanner Gate can detected. Thanks!

**Pull request closed without merging**